job_executor/adapter/local_storage.py

```diff
@@ -291,11 +293,70 @@ def delete_parquet_draft(dataset_name: str) -> None:
     elif os.path.isfile(f'{parquet_path}.parquet'):
         os.remove(f'{parquet_path}.parquet')
 
-def delete_files(file_list : list[str]):
+
+def delete_files(file_list: list[str]):
     """
     Deletes a list of files. Intended to clean up left-over csv files
     * file_list: list[str] - list of csv files to delete
     """
     for file in file_list:
         if os.path.isfile(file):
-            os.remove(file)
\ No newline at end of file
+            os.remove(file)
+
+
+def save_temporary_backup() -> Union[None, LocalStorageError]:
+    """
+    Backs up metadata_all__DRAFT.json, datastore_versions.json and
+    draft_version.json from the datastore to a /tmp directory
+    inside the datastore directory.
+    Raises `LocalStorageError` if /tmp directory already exists.
+    """
+    with open(
+        f'{DATASTORE_DIR}/datastore/datastore_versions.json',
+        encoding='utf-8'
+    ) as f:
+        datastore_versions = json.load(f)
+    with open(
+        f'{DATASTORE_DIR}/datastore/draft_version.json',
+        encoding='utf-8'
+    ) as f:
+        draft_version = json.load(f)
+    with open(
+        f'{DATASTORE_DIR}/datastore/metadata_all__DRAFT.json',
+        encoding='utf-8'
+    ) as f:
+        metadata_all_draft = json.load(f)
+    tmp_dir = Path(DATASTORE_DIR) / 'tmp'
+    if os.path.isdir(tmp_dir):
+        raise LocalStorageError('/tmp directory already exists')
+    os.mkdir(tmp_dir)
+    with open(tmp_dir / 'draft_version.json', 'w', encoding='utf-8') as f:
+        json.dump(draft_version, f)
+    with open(
+        tmp_dir / 'metadata_all__DRAFT.json', 'w', encoding='utf-8'
+    ) as f:
+        json.dump(metadata_all_draft, f)
+    with open(tmp_dir / 'datastore_versions.json', 'w', encoding='utf-8') as f:
+        json.dump(datastore_versions, f)
+
+
+def delete_temporary_backup() -> Union[None, LocalStorageError]:
+    """
+    Deletes the /tmp directory within the datastore if the directory
+    exists. Raises `LocalStorageError` if there are any unrecognized files
+    in the directory.
+    """
+    tmp_dir = Path(DATASTORE_DIR) / 'tmp'
+    if not os.path.isdir(Path(DATASTORE_DIR) / 'tmp'):
```

Comment:
In this case save_temporary_backup() has not been called (tmp does not exist), but delete_temporary_backup() has been called. How can this happen?

Reply:
The file system does not reset itself in between tests, but before and after each test module. So the last test saves the back up, and this one deletes it 👍🏻

Reply:
Can this clean-up-after-tests behavior be moved to tests? Production code should not have to deal with such cases as it introduces some confusion.

Reply:
Agree 👍🏻 Will make it a function level setup/teardown

Reply:
Don't forget to remove the `if` here :)